lib/ansible/modules/files/synchronize.py

```diff
@@ -520,7 +538,27 @@ def main():
     cmd.append(source)
     cmd.append(dest)
     cmdstr = ' '.join(cmd)
-    (rc, out, err) = module.run_command(cmd)
+
+    if not (PY3 and rsync_password):
```

Comment:
The logic here seems like it leaves of PY2 + rsync_password?

Reply:
i've copied it from here: https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/connection/ssh.py#L711

Reply:
i see, that's because `pass_fds` not implemented in python2

Reply:
i am wondering if some warning or error need to be produced to react on such case?

and if so, should this handling be also added to ssh connection plugin?

Reply:
@sivel could you review this code?  I don't know the history but I suspect that PY3 needs the pass_fds parameter and PY2 doesn't but both PY and PY2 need to write the portion which then handles writing to the ssh_pass pipe.

Reply:
@abadger it seems you were the one who added that PY3 branch for `pass_fds`: https://github.com/ansible/ansible/pull/17863